numpy/core/arrayprint.py

```diff
@@ -337,6 +354,29 @@ def _array2string(a, max_line_width, precision, suppress_small, separator=' ',
     return lst
 
 
+def _recursive_guard(fillvalue='...'):
```

Comment:
Good point - I'll fix that up when I find the time (or else, you can if you don't want to wait - this PR is editable)